server/src/main/java/com/objectcomputing/checkins/services/member_skill/MemberSkillController.java

```diff
@@ -56,13 +98,18 @@
      */
 
     @Post(value = "/")
-    public HttpResponse<MemberSkill> createAMemberSkill(@Body @Valid @NotNull MemberSkillCreateDTO memberSkill, HttpRequest<MemberSkillCreateDTO> request) {
-        MemberSkill newMemberSkill = memberSkillsService.save(new MemberSkill(memberSkill.getMemberid(), memberSkill.getSkillid()));
-
-        return HttpResponse
-                .created(newMemberSkill)
-                .headers(headers -> headers.location(
-                        URI.create(String.format("%s/%s", request.getPath(), newMemberSkill.getId()))));
+    public Single<HttpResponse<MemberSkill>> createAMemberSkill(@Body @Valid @NotNull MemberSkillCreateDTO memberSkill, HttpRequest<MemberSkillCreateDTO> request) {
+
+        return Single.fromCallable(() -> memberSkillsService.save(new MemberSkill(memberSkill.getMemberid(),
+                memberSkill.getSkillid())))
+                .observeOn(Schedulers.from(eventLoopGroup))
+                .map(createdMemberSkill -> {
+                    //Using code block rather than lambda so we can log what thread we're in
```

Comment:
This comment was there for demonstrative purposes in the draft PR, but if you remove it you can transform this into a lambda and make much cleaner code.

Reply:
deleted the comment and transformed to lambda.